test/functional/forall/test-forall-rangesegment.hpp

```diff
@@ -0,0 +1,73 @@
+//~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~//
+// Copyright (c) 2016-20, Lawrence Livermore National Security, LLC
+// and RAJA project contributors. See the RAJA/COPYRIGHT file for details.
+//
+// SPDX-License-Identifier: (BSD-3-Clause)
+//~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~//
+
+#ifndef __TEST_FORALL_RANGESEGMENT_HPP__
+#define __TEST_FORALL_RANGESEGMENT_HPP__
+
+#include "RAJA/RAJA.hpp"
+#include "camp/resource.hpp"
+
+template<typename T>
+void ForallRangeSegmentFunctionalTest_host(T first, T last)
+{
+  RAJA::TypedRangeSegment<T> r1(first, last);
+  RAJA::Index_type N = r1.end() - r1.begin();
+
+  camp::resources::Resource res{camp::resources::Host()};
+  T * ref_array = res.allocate<T>(N);
+
+  for(RAJA::Index_type i=0; i < r1.size(); i++)
```

Comment:
If we include a parameter for the index type so we can easily test alternatives; e.g., int, long, etc. does this cause issues? I think we need to test cases representative of how users use these things ans assume that they are not all using RAJA::Index_type.

Reply:
I was using these loops to iterate over the check arrays rather than the actual test so I didn't think it was as important for them to iterate over the testing-type, but I will change this type to the testing-type then it will reflect the testing loop.